src/org/labkey/test/components/glassLibrary/grids/QueryGrid.java

```diff
@@ -52,18 +50,8 @@ public WebDriver getDriver()
 
     public ResponsiveGrid getGrid()
     {
-        _responsiveGrid = new ResponsiveGrid.ResponsiveGridFinder(_driver).findOptional(_queryGridPanel);
-        return _responsiveGrid.orElseThrow();
-    }
-
-    public boolean hasGridBar()
-    {
-        return _gridBar.isPresent();
-    }
-
-    public boolean hasGrid()
-    {
-        return _responsiveGrid.isPresent();
+        _responsiveGrid = new ResponsiveGrid.ResponsiveGridFinder(_driver).find(_queryGridPanel);
+        return _responsiveGrid;
```

Comment:
`_responsiveGrid` was set in the constructor. We should be able to just return it here.
```suggestion
        return _responsiveGrid;
```

Reply:
The call to the finder was there originally (so to speak). The only thing that really changed here was to no longer using optionals. 
I remember we discussed this way back and landed on calling the finder when the grid is requested because it covers the scenario of the grid being updated between the constructor and when this function might be called.
An alternative could be to check for staleness and only call the finder then, otherwise return the grid found int he constructor.